### DISCUSSION TITLE
Why folding not working for rust .rs files but working for .lua files?

### DISCUSSION THREAD

**Discussion by Moanrisy** [LazyVim/LazyVim#2078] [Q&A] [answered]
When I press SPC c s

it show the outline of variable and functions.
this is possbile because treesitter right?

and my config for folding already using treesitter
```
vim.opt.foldmethod = "expr"
vim.opt.foldexpr = "nvim_treesitter#foldexpr()"
-- vim.cmd([[ set nofoldenable]])         -- Disable folding at startup
vim.opt.foldenable = true
```

Just test it using .js files and it work too

after checking with :checkhealth
```

Parser/Features         H L F I J
  - bash                ✓ ✓ ✓ . ✓
  - c                   ✓ ✓ ✓ ✓ ✓
  - diff                ✓ . . . .
  - html                ✓ ✓ ✓ ✓ ✓
  - javascript          ✓ ✓ ✓ ✓ ✓
  - jsdoc               ✓ . . . .
  - json                ✓ ✓ ✓ ✓ .
  - json5               ✓ . . . ✓
  - jsonc               ✓ ✓ ✓ ✓ ✓
  - lua                 ✓ ✓ ✓ ✓ ✓
  - luadoc              ✓ . . . .
  - luap                ✓ . . . .
  - markdown            ✓ . ✓ ✓ ✓
  - markdown_inline     ✓ . . . ✓
  - python              ✓ ✓ ✓ ✓ ✓
  - query               ✓ ✓ ✓ ✓ ✓
  - regex               ✓ . . . .
  - toml                ✓ ✓ ✓ ✓ ✓
  - tsx                 ✓ ✓ ✓ ✓ ✓
  - typescript          ✓ ✓ ✓ ✓ ✓
  - vim                 ✓ ✓ ✓ . ✓
  - vimdoc              ✓ . . . ✓
  - yaml                ✓ ✓ ✓ ✓ ✓
```

it did'nt have rust parser?
I already setup rust using
:lazyExtras

find rust LSP then x to install it

**Comment by Moanrisy** [comment]
Solved by adding
rust in here

  {
    "nvim-treesitter/nvim-treesitter",
    opts = {
      ensure_installed = {

**Reply by dpetka2001** [reply]
You shouldn't have this in your configuration like this. It overrides the `nvim-treesitter` spec and that's why you didn't have the `rust` parser. You should `vim.list_extend` the ensure_installed table of treesitter spec like [this](https://www.lazyvim.org/extras/lang/rust#nvim-treesitter).

**Reply by Moanrisy** [reply]
But it's working now.
Btw I use all the default in plugins/example.lua

Then I write it there without extending.

**Reply by dpetka2001** [reply, marked as answer]
It's working now because you added the `rust` parser in your `treesitter` spec which doesn't extend the default LazyVim treesitter spec but overwrites it. That's the problem. If you use `vim.list_extend` for your `ensure_installed`, you wouldn't have this problem with LazyExtras rust.

Edit: it's also mentioned in the `example.lua` file
> -- since `vim.tbl_deep_extend`, can only merge tables and not lists, the code above
  -- would overwrite `ensure_installed` with the new value.
  -- If you'd rather extend the default config, use the code below instead:
  {
    "nvim-treesitter/nvim-treesitter",
    opts = function(_, opts)
      -- add tsx and treesitter
      vim.list_extend(opts.ensure_installed, {
        "tsx",
        "typescript",
      })
    end,
  },

**Reply by Moanrisy** [reply]
Oh I see. I should have read it not just skim it.
Ok thanks for the tips 🙏

So the problem why I didn't get rust parser although I install with :lazyExtras is because I overwrite it.


I will try reinstall with fresh config later and make sure rust parser automatically got installed with :lazyExtras
